packages/angular/src/generators/ng-add/utilities/workspace.ts

```diff
@@ -0,0 +1,350 @@
+  try {
+    if (!tree.exists('package.json')) {
+      throw new Error('Cannot find package.json');
+    }
+
+    if (!tree.exists('angular.json')) {
+      throw new Error('Cannot find angular.json');
+    }
+
+    const e2eKey = getE2eKey(projects);
+    const e2eApp = getE2eProject(projects);
+
+    if (!e2eApp) {
+      return;
+    }
+
+    if (isProtractorE2eProject(e2eApp.config)) {
+      if (tree.exists(e2eApp.config.targets.e2e.options.protractorConfig)) {
+        return;
+      }
+
+      console.info(
+        `Make sure the "${e2eKey}.architect.e2e.options.protractorConfig" is valid or the "${e2eKey}" project is removed from "angular.json".`
+      );
+      throw new Error(
+        `An e2e project with Protractor was found but "${e2eApp.config.targets.e2e.options.protractorConfig}" could not be found.`
+      );
+    }
+
+    if (isCypressE2eProject(e2eApp.config)) {
+      const configFile = getCypressConfigFile(e2eApp.config);
+      if (configFile && !tree.exists(configFile)) {
+        throw new Error(
+          `An e2e project with Cypress was found but "${configFile}" could not be found.`
+        );
+      }
+
+      if (!tree.exists('cypress')) {
+        throw new Error(
+          `An e2e project with Cypress was found but the "cypress" directory could not be found.`
+        );
+      }
+
+      return;
+    }
+
+    throw new Error(
+      `An e2e project was found but it's using an unsupported executor "${e2eApp.config.targets.e2e.executor}".`
+    );
+  } catch (e) {
+    console.error(e.message);
+    console.error(
+      'Your workspace could not be converted into an Nx Workspace because of the above error.'
+    );
+    throw e;
+  }
+}
+
+export function createNxJson(
+  tree: Tree,
+  options: GeneratorOptions,
+  setWorkspaceLayoutAsNewProjectRoot: boolean = false
+): void {
+  const { projects = {}, newProjectRoot = '' } = readJson(tree, 'angular.json');
+  // TODO: temporarily leaving this here because it's the old behavior for a
+  // minimal migration, will be removed in a later PR
+  const hasLibraries = Object.keys(projects).find(
+    (project) =>
+      projects[project].projectType &&
+      projects[project].projectType !== 'application'
+  );
+  if (Object.keys(projects).length !== 1 || hasLibraries) {
+    throw new Error(
+      `The schematic can only be used with Angular CLI workspaces with a single application.`
+    );
+  }
+
+  writeJson<NxJsonConfiguration>(tree, 'nx.json', {
+    npmScope: options.npmScope,
+    affected: {
+      defaultBase: options.defaultBase ?? deduceDefaultBase(),
+    },
+    implicitDependencies: {
+      'package.json': {
+        dependencies: '*',
+        devDependencies: '*',
+      },
+      '.eslintrc.json': '*',
+    },
+    tasksRunnerOptions: {
+      default: {
+        runner: 'nx/tasks-runners/default',
+        options: {
+          cacheableOperations: ['build', 'lint', 'test', 'e2e'],
+        },
+      },
+    },
+    targetDependencies: {
```

Comment:
Victor mentioned that this might be changed, so it's something we should just keep an eye on